streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamTransformValues.java

```diff
@@ -46,8 +58,104 @@ public KStreamTransformValuesProcessor(ValueTransformer<V, R> valueTransformer)
         }
 
         @Override
-        public void init(ProcessorContext context) {
-            valueTransformer.init(context);
+        public void init(final ProcessorContext context) {
+            valueTransformer.init(
+                new ProcessorContext() {
+                    @Override
+                    public String applicationId() {
+                        return context.applicationId();
+                    }
+
+                    @Override
+                    public TaskId taskId() {
+                        return context.taskId();
+                    }
+
+                    @Override
+                    public Serde<?> keySerde() {
+                        return context.keySerde();
+                    }
+
+                    @Override
+                    public Serde<?> valueSerde() {
+                        return context.valueSerde();
+                    }
+
+                    @Override
+                    public File stateDir() {
+                        return context.stateDir();
+                    }
+
+                    @Override
+                    public StreamsMetrics metrics() {
+                        return context.metrics();
+                    }
+
+                    @Override
+                    public void register(final StateStore store, final boolean loggingEnabled, final StateRestoreCallback stateRestoreCallback) {
```

Comment:
I had a similar though about this, but this should go into its own PR. The `forward` issues is only for `ValueTransformer` while the issue you mention is for all processor nodes and we should fix them all in one PR.